internal/transport/http2_client.go

```diff
@@ -176,7 +176,9 @@ func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error
 	if networkType == "tcp" && useProxy {
 		return proxyDial(ctx, address, grpcUA)
 	}
-	return (&net.Dialer{}).DialContext(ctx, networkType, address)
+	// KeepAlive is set to a negative value to prevent Go's override of the TCP
+	// keepalive time and interval; retain the OS default.
+	return (&net.Dialer{KeepAlive: time.Duration(-1)}).DialContext(ctx, networkType, address)
```

Comment:
Looks like we also need to fix:

https://github.com/grpc/grpc-go/blob/e88e8498c6df39703d1992bc4b677e7ef477145d/internal/transport/proxy.go#L125

Reply:
@dfawley do you want to do the same for the test and benchmark files with dialers that use TCP?

Reply:
Do the benchmarks use custom dialers?  If so, then I'd say "yes, please, if you don't mind".  Tests should be fine to keep as-is.